MIDAS/src/systems.cpp

```diff
@@ -28,9 +28,27 @@ DECLARE_THREAD(logger, RocketSystems* arg) {
 }
 
 DECLARE_THREAD(barometer, RocketSystems* arg) {
+    // Reject single rogue barometer readings that are very different from the immediately prior reading
+    // Will only reject a certain number of readings in a row
+    Barometer prev_reading;
+    constexpr float altChgThreshold = 200; // meters?
+    constexpr float presChgThreshold = 500; // milibars
+    constexpr float tempChgThreshold = 10; // degrees C?
+    constexpr unsigned int maxConsecutiveRejects = 5;
```

Comment:
Is 5 a reasonable number here? Barometer anomalies are most likely not going to last for 5 ticks (and, as we saw in sg1.3, will probably only last one tick). I would rather in the worst case get 1 in 2 (or 1 in 3) sensor readings rather than 1 in 5.

Reply:
Well we update at about 166 Hz, so 5 ticks will only be 30 ms. Not sure if that will affect our readings too much, and I think it's better to err on the side of caution

Reply:
I would imagine it's unlikely to see that many garbage data points. I would change maxConsecutiveRejects to 2 or 3.